src/Datadog.Trace/Configuration/StringConfigurationSource.cs

```diff
@@ -89,7 +90,7 @@ public abstract class StringConfigurationSource : IConfigurationSource
         {
             string value = GetString(key);
 
-            return double.TryParse(value, out double result)
+            return double.TryParse(value, NumberStyles.Any, CultureInfo.InvariantCulture, out double result)
```

Comment:
`NumberStyles.Any` allows syntax like `1e2` or `(1)`. Are we ok with those? I would think we only want to allow leading and trailing whitespace, decimal point, and maybe leading sign if we even accept negative values.

Reply:
Should we add this to `int.TryParse()` above for consistency? Not sure that it will have any effect, though, since that one should be a string of digits only.